controllers/grid/queries/form/QueryForm.php

```diff
@@ -327,14 +327,11 @@ public function fetch($request, $template = null, $display = false, $actionArgs
             // if current user is an anonymous reviewer, filter out authors
             foreach ($reviewAssignments as $reviewAssignment) {
                 if ($reviewAssignment->getReviewerId() == $user->getId()) {
-                    if ($reviewAssignment->getReviewMethod() != ReviewAssignment::SUBMISSION_REVIEW_METHOD_OPEN) {
-                        // Replaces StageAssignmentDAO::getBySubmissionAndRoleId
-                        $excludeUsers = StageAssignment::withSubmissionIds([$query->assocId])
+                    if ($reviewAssignment->getReviewMethod() == ReviewAssignment::SUBMISSION_REVIEW_METHOD_DOUBLEANONYMOUS) {
+                        $stageAssignments = StageAssignment::withSubmissionIds([$query->assocId])
                             ->withRoleIds([Role::ROLE_ID_AUTHOR])
-                            ->withUserId($user->getId())
-                            ->get()
-                            ->pluck('userId')
-                            ->all();
+                            ->get();
+                        $excludeUsers = $stageAssignments->pluck('userId')->all();
```

Comment:
Can we retrieve excluded users without intermediate variables? Like:
```php
$excludeUsers = StageAssignment::withSubmissionIds([$query->assocId])
    ->withRoleIds([Role::ROLE_ID_AUTHOR])
    ->get()
    ->pluck('userId')
    ->all();
    ```

Reply:
I'll update the code as you suggested here. But as mentioned in the  [comment](https://github.com/pkp/pkp-lib/pull/10391#discussion_r1785248281) about the accosser, I'll proceed with the final update once we decide on the approach regarding the accessor.